imports/plugins/included/taxes-avalara/client/components/avalaraSettingsForm.js

```diff
@@ -0,0 +1,182 @@
+ * @method AvalaraSettingsForm
+ * @summary renders a form for updating Avalara settings.
+ * @param {Object} props - some data for use by this component.
+ * @property {Function} handleSubmit - a function for saving new Avalara settings.
+ * @property {Array} hiddenFields - the fields (of the Avalara Package) to hide from the form.
+ * @property {Object} settings - the value of the "settings" field in the Avalara Package.
+ * @property {Object} shownFields - info about the fields the form is to show.
+ * @return {Node} - a React node containing the Avalara settings form.
+ */
+class AvalaraSettingsForm extends Component {
+  /**
+  * @name AvalaraSettingsForm propTypes
+  * @type {propTypes}
+  * @param {Object} props - React PropTypes
+  * @property {Object} fieldsProps - map of field specific properties for avalara settings.
+  * @property {Function} handleSubmit - a function that saves new Avalara settings.
+  * @property {Array} hiddenFields - an array of the Avalara Package's fields
+  * to hide from the settings form.
+  * @property {Object} logFieldsProps - map of field specific properties for logs.
+  * @property {Object} settings - the value of the "settings" field in the Avalara Package.
+  * @property {Object} shownFields - info about the fields of the Avalara Package
+  * that the settings form will allow users to change.
+  * @property {Object} shownLogFields - fields to show from Log schema
+  * @return {Array} React propTypes
+  */
+  static propTypes = {
+    fieldsProp: PropTypes.object,
+    handleSubmit: PropTypes.func,
+    handleTestCredentials: PropTypes.func,
+    hiddenFields: PropTypes.arrayOf(PropTypes.string),
+    logFieldsProp: PropTypes.object,
+    settings: PropTypes.object,
+    shownFields: PropTypes.object,
+    shownLogFields: PropTypes.object
+  };
+  static filteredFields = [ "data.request.data.date", "data.request.data.type" ];
+  static noDataMessage = i18next.t("logGrid.noLogsFound");
+
+  // helper adds a class to every grid row
+  static customRowMetaData = {
+    bodyCssClassName: () =>  {
+      return "log-grid-row";
+    }
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editingId: undefined,
+      showLogs: this.props.settings.avalara.enableLogging,
+      log: undefined,
+      fieldsProp: assign(clone(props.fieldsProp), {
+        "settings.avalara.enableLogging": {
+          handleChange: this.handleLogToggle
+        }
+      })
+    };
+  }
+
+  handleLogToggle = () => {
+    this.setState(() => ({ showLogs: !this.state.showLogs }));
+  }
+
+  handleTestCredentials = (event) => {
+    this.props.handleTestCredentials(event, this.props.settings);
+  }
+
+  editRow = (options) => {
+    const currentId = this.state.editingId;
+    this.setState(() => {
+      const id = options.props.data._id;
+      const log = Logs.findOne(id) || {};
+      log.data = JSON.stringify(log.data, null, 4);
+      return {
+        editingId: id,
+        log
+      };
+    });
```

Comment:
Why passing a function to setState rather than just having those lines before the `setState` call?